lib/online_order.rb

```diff
@@ -0,0 +1,106 @@
+require_relative '../lib/order'
+
+require 'csv'
+require 'awesome_print'
+
+module Grocery
+  class  OnlineOrder < Grocery::Order
+    attr_reader :customerid, :status
+
+    #overwrites initialize method for Order
+    def initialize(id, products, customerid, status)
+      super(id, products)
+      @customerid = customerid
+      @status = status
+    end
+
+    #returns a list of OnlineOrder instances where the value of the customer's ID matches the passed parameter.
+    def self.find_by_customer(customer_id)
+      online_orders_csv = CSV.read("support/online_orders.csv", 'r',headers: true).to_a
+
```

Comment:
Instead of calling `CSV.read` here, you should get the list of orders by calling `.all`.